src/Core/Application/RealTime/UseCase/FindHostCategory/FindHostCategory.php

```diff
@@ -0,0 +1,121 @@
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ImcompatibilityResponse;
+use Core\Application\RealTime\Repository\ReadTagRepositoryInterface;
+use Core\Infrastructure\RealTime\Api\FindHostCategory\FindHostCategoryPresenter;
+
+class FindHostCategory
+{
+    use LoggerTrait;
+
+    private const MINIMUM_BBDO_VERSION_SUPPORTED = '3.0.0',
+                  BBDO_VERSION_CONFIG_KEY = 'bbdo_version';
+
+    /**
+     * @param ReadTagRepositoryInterface $repository
+     */
+    public function __construct(
+        private ReadTagRepositoryInterface $repository,
+        private BrokerRepositoryInterface $brokerRepository
+    ) {
+    }
+
+    /**
+     * @param FindHostCategoryPresenterInterface $presenter
+     */
+    public function __invoke(FindHostCategoryPresenterInterface $presenter): void
+    {
+        $this->info('Searching for host categories');
+
+        try {
+            $hostCategories = $this->repository->findAllByTypeId(Tag::HOST_CATEGORY_TYPE_ID);
+            if (empty($hostCategories)) {
+                if (! $this->isBBDOVersionCompatible()) {
+                    $this->handleImcompatibleBBDOVersion($presenter);
+                    return;
+                }
+            }
+        } catch (\Throwable $e) {
+            $this->error(
+                'An error occured while retrieving host categories',
+                [
+                    'trace' => $e->getTraceAsString()
+                ]
+            );
+            $presenter->setResponseStatus(new ErrorResponse('An error occured while retrieving host categories'));
+            return;
+        }
+
+        $presenter->present(
+            $this->createResponse($hostCategories)
+        );
+    }
+
+    /**
+     * @param Tag[] $categories
+     * @return FindHostCategoryResponse
+     */
+    private function createResponse(array $categories): FindHostCategoryResponse
+    {
+        return new FindHostCategoryResponse($categories);
+    }
+
+    /**
+     * @param FindHostCategoryPresenterInterface $presenter
+     * @return void
+     */
+    private function handleImcompatibleBBDOVersion(FindHostCategoryPresenterInterface $presenter): void
+    {
+        $message = 'BBDO protocol version enabled not compatible with this feature. Version needed '
+            . self::MINIMUM_BBDO_VERSION_SUPPORTED . ' or higher';
+        $this->error($message);
+        $presenter->setResponseStatus(new ImcompatibilityResponse($message));
+    }
+
+    /**
+     * Checks if at least on monitoring server has BBDO protocol in version 3.0.0
+     *
+     * @return boolean
+     */
+    private function isBBDOVersionCompatible(): bool
+    {
+        $brokerConfigurations = $this->brokerRepository->findAllByParameterName(self::BBDO_VERSION_CONFIG_KEY);
+        foreach ($brokerConfigurations as $brokerConfiguration) {
+            if (
+                version_compare(
```

Comment:
No check on the 'daemon' property ?

Reply:
I guess not, as the method `findAllByParameterName` returns only a `BrokerConfiguration` entity which represent only a subset of the configuration (one key and one value).

To get this parameter, an other call to `findAllByParameterName` needs to be done. Do we need to check for the daemon property ? IDK